test/new_relic/agent/configuration/default_source_test.rb

```diff
@@ -291,6 +285,81 @@ def test_allowlist_blocks_invalid_values_and_uses_a_default
       end
     end
 
+    def test_boolean_configs_accepts_yes_on_and_true_as_strings
+      key = :'send_data_on_exit'
+      default = ::NewRelic::Agent::Configuration::DefaultSource.default_for(key)
+      config_array = %w[yes on true]
+
+      config_array.each do |value|
+        with_config(key => value) do
+          assert NewRelic::Agent.config[key]
+        end
+      end
+    end
+
+    def test_boolean_configs_accepts_yes_on_and_true_as_symbols
+      key = :'send_data_on_exit'
+      default = ::NewRelic::Agent::Configuration::DefaultSource.default_for(key)
+      config_array = %i[yes on true]
+
+      config_array.each do |value|
```

Comment:
For all of these tests, given that you only ever iterate through the array and don't interact with the array in any other manner, you could optionally forego bothering to store it in a variable:

```suggestion
      %i[yes on true].each do |value|
```

Reply:
Love this- accepted! https://github.com/newrelic/newrelic-ruby-agent/pull/2847/commits/be01eafceca9862380560b212346bac17f6db246